rmw_connextdds_common/src/ndds/rmw_type_support_ndds.cpp

```diff
@@ -268,6 +277,44 @@ RMW_Connext_TypePlugin_destroy_data(void ** sample, void * user_data)
   delete msg;
 }
 
+static
+RTIBool
+RMW_Connext_TypePlugin_create_key(void ** key, void * user_data)
+{
+  RMW_Connext_MessageTypeSupport * const type_support =
+    reinterpret_cast<RMW_Connext_MessageTypeSupport *>(user_data);
+
+  const auto buffer_size = (type_support->unbounded_key()) ?
+    0 : type_support->type_key_serialized_size_max();
+
+  RMW_Connext_Message * msg = new (std::nothrow) RMW_Connext_Message();
+  if (nullptr == msg) {
+    return RTI_FALSE;
+  }
+  if (RMW_RET_OK != RMW_Connext_Message_initialize(msg, type_support, buffer_size)) {
+    delete msg;
+    return RTI_FALSE;
+  }
+  *key = msg;
+
+  return RTI_TRUE;
+}
+
+static
+void
+RMW_Connext_TypePlugin_destroy_key(void ** key, void * user_data)
```

Comment:
(not blocking PR, just a suggestion) i would do the following to keep the code cleaner, if that is unused for now.

```suggestion
RMW_Connext_TypePlugin_destroy_key(void ** key, [[maybe_unused]] void * user_data)
```

Reply:
That's a really nice addition! I'll make sure to slowly clean up the code after this is merged